archiver/main.go

```diff
@@ -504,162 +488,32 @@ func manualRun() {
-	CategoryID int    `json:"category_id"`
-	Name       string `json:"name"`
-}) string {
-	categoryNames := make([]string, len(categories))
-	for i := range categories {
-		categoryNames[i] = categories[i].Name
-	}
-	return strings.Join(categoryNames, ",")
-}
-
-func flattenRecipients(recipients []KnowBe4Recipient) ([]KnowBe4FlatRecipient, error) {
-	flatRecipients := make([]KnowBe4FlatRecipient, len(recipients))
-	for i, recipient := range recipients {
-		flatRecipient, err := flattenRecipient(recipient)
-		if err != nil {
-			return flatRecipients, err
-		}
-		flatRecipients[i] = flatRecipient
-	}
-	return flatRecipients, nil
-}
-
-func flattenRecipient(recipient KnowBe4Recipient) (KnowBe4FlatRecipient, error) {
-	var flatRecipient KnowBe4FlatRecipient
-	if err := ConvertToOtherType(recipient, &flatRecipient); err != nil {
-		return flatRecipient, err
-	}
-
-	flatRecipient.UserID = recipient.User.ID
-	flatRecipient.UserActiveDirectoryGUID = recipient.User.ActiveDirectoryGUID
-	flatRecipient.UserFirstName = recipient.User.FirstName
-	flatRecipient.UserLastName = recipient.User.LastName
-	flatRecipient.UserEmail = recipient.User.Email
-	flatRecipient.TemplateID = recipient.Template.ID
-	flatRecipient.TemplateName = recipient.Template.Name
-
-	return flatRecipient, nil
-}
-
-func flattenCampaigns(campaigns []KnowBe4Campaign) ([]KnowBe4FlatCampaign, error) {
-	flatCampaigns := make([]KnowBe4FlatCampaign, len(campaigns))
-	for i, c := range campaigns {
-		flatCampaign, err := flattenCampaign(c)
-		if err != nil {
-			return flatCampaigns, err
-		}
-		flatCampaigns[i] = flatCampaign
-	}
-	return flatCampaigns, nil
-}
-
-func flattenCampaign(campaign KnowBe4Campaign) (KnowBe4FlatCampaign, error) {
-	var flatCampaign KnowBe4FlatCampaign
-	if err := ConvertToOtherType(campaign, &flatCampaign); err != nil {
-		return flatCampaign, err
-	}
-
-	flatCampaign.Groups = flattenGroupSummaries(campaign.Groups)
-	flatCampaign.DifficultyFilter = flattenIntSlice(campaign.DifficultyFilter)
-	flatCampaign.Psts = flattenPstSlice(campaign.Psts)
-
-	return flatCampaign, nil
-}
-
-func flattenIntSlice(intSlice []int) string {
-	stringSlice := make([]string, len(intSlice))
-	for i := range intSlice {
-		stringSlice[i] = strconv.Itoa(intSlice[i])
+// marshalJsonLines is a partial implementation of JSON Lines
+// It only supports lists of objects. It will not generate the simplified case with no nested objects
+// as shown in the first example on https://jsonlines.org
+func marshalJsonLines(input interface{}) ([]byte, error) {
+	if input == nil {
+		return nil, fmt.Errorf("marshalJsonLines nil input")
 	}
-	return strings.Join(stringSlice, ",")
-}
-
-func flattenPstSlice(pstSlice []PstSummary) string {
-	stringSlice := make([]string, len(pstSlice))
-	for i := range pstSlice {
-		stringSlice[i] = strconv.Itoa(pstSlice[i].PstId)
+	list, ok := input.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("marshalJsonLines input is not []interface{}")
```

Comment:
An alternative here is to assume we're just adding one row and create a new []interface{} instead with it as a value.